generators/chipyard/src/main/scala/config/SpikeConfigs.scala

```diff
@@ -27,7 +27,7 @@ class SpikeFastUARTConfig extends Config(
 class SpikeUltraFastConfig extends Config(
   new chipyard.WithSpikeTCM ++
   new chipyard.WithNSpikeCores(1) ++
-  new chipyard.config.WithUARTFIFOEntries(128, 128) ++
+  new chipyard.config.WithUART(txEntries=128, rxEntries=128) ++
```

Comment:
This will add another UART to the system with the parameters that you care about (since the `WithUART` fragment is additive instead of overriding the original UART params). Thus the original `WithUARTFIFOEntries` is needed.

Reply:
Updated. I'm not sure if this approach works...

Reply:
I don't think removing the UART from these designs is the right thing to do. 

If we just change the default number of entries for all designs, wouldn't this solve the problem with minimal side effects?

Reply:
Using 128-entry FIFO is unrealistic, and would not be applicable for FPGA and ASIC tapeout designs.

Reply:
I think sagar mentioned that this is a special case that we need a huge FIFO for UART.

Reply:
Right yeah. For simulation we need large FIFOs for responsiveness. 

We should isolate the many-fifo-entries to only the simulation configs, mainly the FireSim configs and the Spike tile configs.

Reply:
Agreed. The changed code still doesn't work now since you are removing all UARTs from the design (I assume the Spike configs need some UART)

Reply:
Oh sorry, I messed up the order of the config fragments.

Reply:
fixed.

Reply:
Thanks. Now the only thing left is to fix the Firesim configs, which need 128+ FIFO entries. You can just put `WithUARTFIFOEntries` in `WithFiresimConfigTweaks`.

Reply:
fixed